packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -0,0 +1,391 @@
+/// will cascade down to other dependent parameters to create a cohesive
+/// visual effect. For instance, if a [primaryColor] is specified, it would
+/// cascade down to affect some fonts in [textTheme] if [textTheme] is not
+/// specified.
+///
+/// When retrieved using [CupertinoTheme.of], reading each property will create
+/// a dependency in the [BuildContext] given to [CupertinoTheme.of]. The
+/// given [BuildContext] will rebuild when that property changes.
+///
+/// See also:
+///
+///  * [CupertinoTheme], in which this [CupertinoThemeData] is inserted.
+///  * [ThemeData], a Material equivalent that also configures Cupertino
+///    styling via a [CupertinoThemeData] subclass [MaterialBasedCupertinoThemeData].
+@immutable
+class CupertinoThemeData extends Diagnosticable {
+  /// Create a [CupertinoTheme] styling specification.
+  ///
+  /// Unspecified parameters can be derived from another parameter or default
+  /// to an iOS default style.
+  const CupertinoThemeData({
+    Brightness brightness,
+    Color primaryColor,
+    Color primaryContrastingColor,
+    CupertinoTextTheme textTheme,
+    Color barBackgroundColor,
+    Color scaffoldBackgroundColor,
+    Color tableBackgroundColor,
+  }) : this.raw(
+        brightness,
+        primaryColor,
+        primaryContrastingColor,
+        textTheme,
+        barBackgroundColor,
+        scaffoldBackgroundColor,
+        tableBackgroundColor,
+      );
+
+  /// A raw constructor used by subclasses to populate the superclass's fields
+  /// to get superclass behaviors not overriden by subclasses.
+  ///
+  /// Same as the default constructor but with positional arguments to avoid
+  /// forgetting any.
+  @protected
+  const CupertinoThemeData.raw(
+    this._brightness,
+    this._primaryColor,
+    this._primaryContrastingColor,
+    this._textTheme,
+    this._barBackgroundColor,
+    this._scaffoldBackgroundColor,
+    this._tableBackgroundColor,
+  );
+
+  bool get _isLight => brightness == Brightness.light;
+
+  final Brightness _brightness;
+  /// The general brightness theme of the [CupertinoThemeData].
+  ///
+  /// Affects all other theme properties when unspecified. Defaults to
+  /// [Brightness.light].
+  ///
+  /// If coming from a Material [Theme] and unspecified, [brightness] will be
+  /// derived from the Material [ThemeData]'s `brightness`.
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Brightness get brightness => _brightness ?? Brightness.light;
+
+  final Color _primaryColor;
+  /// A color used on interactive elements of the theme.
+  ///
+  /// This color is generally used on text and icons in buttons and tappable
+  /// elements. Defaults to blue or orange when [brightness] is light or dark.
+  ///
+  /// If coming from a Material [Theme] and unspecified, [primaryColor] will be
+  /// derived from the Material [ThemeData]'s `colorScheme.primary`. However, in
```

Comment:
I will not belabor this trivial point any further but: "derived from" sounds more complicated than simple copying.  In this case "will be derived from" could just be "the same as".

Reply:
Ya same as below. It doesn't copy it. The actual value at that node is null. It defers the value to the ancestor material theme.